### ISSUE TITLE
Ran into overflow and bounds detection issue

### ISSUE THREAD

**Issue by yasoob** [maxmarcon/live_select#43]
I decided to test live_select but ran into two issues.

## Select options overflow 

This screenshot is from the [demo app](https://live-select.fly.dev/):

<img width="885" alt="image" src="https://github.com/maxmarcon/live_select/assets/3696393/e5b1fc66-6791-4c91-b608-965a08bc27bb">

Any idea why this is overflowing? This happens on both Firefox and Chrome.

## No bounds detection

Most JS-based select dropdowns have some sort of bounds detection. This makes it such that if there is not enough space to show the dropdown below the input field, the dropdown opens upwards. Live_select doesn't appear to support this:

<img width="853" alt="image" src="https://github.com/maxmarcon/live_select/assets/3696393/e4210d66-b7b8-424c-9877-9fb1dfed1a68">

You can look at Headless-UI or [shadcn/ui](https://ui.shadcn.com/docs/components/select) to get a sense of what I am talking about:

<img width="303" alt="image" src="https://github.com/maxmarcon/live_select/assets/3696393/d6129818-72c9-45c4-b504-c7daee50cb69">

Thank you so much for making this open source! :)

**Comment by maxmarcon** [comment]
Thanks for pointing this out.

The second problem (bounds detection) is something I haven't addressed yet because it seems very hard and other things were more important at the time. A PR would be very welcome! 

Regarding the first problem (overflow) : I believe this was introduced by one of my recent changes. I bet if you try `1.1.1` this won't happen.

I'll be able to fix this in a couple of weeks (I'm traveling without my laptop right now) 

Until then, you should be good downgrading to `1 1.1`

**Comment by maxmarcon** [comment]
Until a release with the bugfix is published, the issue can also be fixed by assigning a `relative` position to the live select div container. If one is using tailwind, this is as simple as doing this:

```
<.live_select
   field={my_form[:city_search]}
   container_extra_class="relative" />
```

see [live example](https://live-select.fly.dev/?allow_clear=false&container_extra_class=relative&debounce=100&disabled=false&mode=single&placeholder=Search+for+a+city&search_delay=10&show_styles=true&style=tailwind&style_filter=cont&update_min_len=3&user_defined_options=false)

**Comment by yasoob** [comment]
Thanks for checking in on this Max!

**Comment by maxmarcon** [comment]
Overflow problem has been fixed in [1.2.1](https://hex.pm/packages/live_select/1.2.1)

I added a [section to the cheatsheet](https://hexdocs.pm/live_select/1.2.1/cheatsheet.html#dropdown-that-opens-above-the-input-field) that shows how to open the dropdown above the input field. The "bounds detection" however will have to happen in your live view or live component. Doesn't completely address your issue I know, but it's a steps in the right direction.
